StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Views/PromoBadgeView.swift

```diff
@@ -14,21 +14,32 @@ class PromoBadgeView: UIView {
     private let labelBackground = UIView()
     private let label = UILabel()
     
+    private var appearance: PaymentSheet.Appearance
+    private var cornerRadius: CGFloat?
+    private var text: String
+    private var eligible: Bool
+    
+    override var intrinsicContentSize: CGSize {
+        CGSize(
+            width: labelBackground.layoutMargins.left + label.intrinsicContentSize.width + labelBackground.layoutMargins.right,
+            height: labelBackground.layoutMargins.top + label.intrinsicContentSize.height + labelBackground.layoutMargins.bottom
+        )
+    }
+    
     init(
         appearance: PaymentSheet.Appearance,
         cornerRadius: CGFloat? = nil,
         tinyMode: Bool,
         text: String? = nil
     ) {
+        self.appearance = appearance
+        self.cornerRadius = cornerRadius
+        self.eligible = true
+        self.text = text ?? ""
```

Comment:
Just curious; Any reason to have a default value here rather than keeping the `self.text` field optional?

Reply:
Seemed easier like this, but I reworked it a bit to make it nullable.